### PR TITLE
[IMP] account: improve journal entries recording

### PR DESCRIPTION
**Pull request by william-andre**
Task 1943019
onchange should ensure that you don't have both a credit and a debit <>0
prefill account/partner if the user is recording several lines a row with the same account/partner
the journal should be set by default

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/account/models/account_move.py [hunk 1]
```diff
@@ -22,6 +22,13 @@ class AccountMove(models.Model):
     _order = 'date desc, id desc'
     _inherit = ['mail.thread', 'mail.activity.mixin']
 
+    @api.model
+    def default_get(self, fields):
+        rec = super(AccountMove, self).default_get(fields)
+        if not rec.get('journal_id'):
+            rec.update({'journal_id': self.env['account.journal'].search([('type', '=', 'general')], limit=1).id})
```
Comment (fmdl) [inline addons/account/models/account_move.py]: use current company_id to find the journal.
@qdp-odoo

Comment (qdp-odoo) [reply addons/account/models/account_move.py]: @william-andre indeed. 

Can you also redo a valid commit message? I just ditched it by mistake while force-pushing the branch. Sorry about that

Comment (william-andre) [reply addons/account/models/account_move.py]: @qdp-odoo done, and done

File: addons/account/models/account_move.py [hunk 2]
```diff
@@ -497,6 +504,14 @@ def onchange_tax_ids_create_aml(self):
         for line in self:
             line.recompute_tax_line = True
 
+    @api.onchange('debit', 'credit')
+    def onchange_debit_credit(self):
+        for line in self:
```
Comment (fmdl) [inline addons/account/models/account_move.py]: this logic is not good :

start: debit : 0, credit :0
write in debit : 10
result : debit : 10, credit :0
write in **credit : 5**
result : debit : 10, **credit :0** --> strange

normal behavhior :
result : debit : 0, credit :1

You need to use two on_change

```python
@api.onchange('debit')
def _onchange_debit(self) #use private function
    self.ensure_one()
    if self.debit '!=' 0:
          self.credit = 0

@api.onchange('credit')
def _onchange_credit(self) #use private function
    self.ensure_one()
    if self.credit '!=' 0:
          self.debit = 0
```

@qdp-odoo

Comment (william-andre) [reply addons/account/models/account_move.py]: Yup, this is *exactly* what i had before the force push

Comment (qdp-odoo) [reply addons/account/models/account_move.py]: :-D

Comment (qdp-odoo) [reply addons/account/models/account_move.py]: Ok, ok


and _now_ I understand why we'd need 2 onchanges. My apologize @william-andre you can force-pushback ^^

Comment (fmdl) [reply addons/account/models/account_move.py]: @qdp-odoo it is possible with 1 onchange, but you need to use `self._origin` to know witch fields change.

**Comment by qdp-odoo** [comment]
@robodoo r+

**Pull request closed without merging**